src/compiler/checker.ts

```diff
@@ -5722,7 +5725,14 @@ namespace ts {
             }
             setStructuredTypeMembers(type, members, emptyArray, emptyArray, stringIndexInfo, undefined);
 
-            function addMemberForKeyType(t: Type, propertySymbol?: Symbol) {
+            function addMemberForKeyType(t: Type, propertySymbolOrIndex?: Symbol | number) {
+                let propertySymbol: Symbol;
+                // forEachType offloads to forEach, which calls with a numeric second argument

```

Comment:
I can't comment on line 5724 because of github's limitations, but I would just change that to `foreachType(iterationType, (p,i) => addMemberForKeyType(p))` and remove this whole changed section.

Reply:
We usually avoid inline lambdas, from what I've been told, since they don't get optimized as well as a named function. Could be false, I dunno.